telegram_export/__main__.py

```diff
@@ -271,13 +271,28 @@ def find_dialog(dialogs, query, top=25, threshold=0.7):
         config['Dumper']['OutputDirectory'],
         config['TelegramAPI']['SessionName']
     )
+    """
     client = await (TelegramClient(
         absolute_session_name,
         config['TelegramAPI']['ApiId'],
         config['TelegramAPI']['ApiHash'],
         loop=loop,
         proxy=proxy
     ).start(config['TelegramAPI']['PhoneNumber']))
+    """
+    try:
+        client = await (TelegramClient(
+            absolute_session_name,
+            config['TelegramAPI']['ApiId'],
+            config['TelegramAPI']['ApiHash'],
+            loop=loop,
+            proxy=proxy
+        ).connect()
+        if not client.is_user_authorized():
+                client.send_code_request(config['TelegramAPI']['PhoneNumber'])
+                client.sign_in(config['TelegramAPI']['PhoneNumber'], input('Enter code: '))
+    except SessionPasswordNeededError:
+        client.sign_in(password=config['TelegramAPI']['SecondFactorPassword'])
```

Comment:
[`.start()`](https://telethon.readthedocs.io/en/stable/telethon.client.html#telethon.client.auth.AuthMethods.start) supports a `password` parameter so this is not necessary.